server/src/api/post.rs

```diff
@@ -238,21 +238,13 @@ impl Perform<GetPostsResponse> for Oper<GetPosts> {
     let type_ = ListingType::from_str(&data.type_)?;
```

Comment:
Wow, this looks great! I'll have to make issues to make builders like these for the other telescoping queries with lots of optionals. `CommentView` `ReplyView` are 2 others I can think of, but we can put those in different PRs.